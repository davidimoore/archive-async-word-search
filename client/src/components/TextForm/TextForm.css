.TextFormContainer {
  display: flex;
  width: 35%;
}

.TextFormContainer .TextForm {
  width: 100%;
}

.TextFormContainer .TextForm .WordTextInput {
  border: 1px solid cornflowerblue;
  border-radius: 0;
  padding: 1em;
  width: 80%;
  -webkit-appearance: none;
}

.TextFormContainer .TextForm .Submit {
  border: 1px solid cornflowerblue;
  border-radius: 0;
  padding: 1em;
  -webkit-appearance: none;
}
