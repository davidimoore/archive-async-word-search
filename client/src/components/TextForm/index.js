import React, { Component } from 'react';
import './TextForm.css';

export default class TextForm extends Component {
  state = { words: "" };

  handleChange = (event) => {
    this.setState({ words: event.target.value })
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { submitHandler } = this.props;
    const { words } = this.state;
    submitHandler(words)
  };

  render() {
    return (
      <div
        className={ `TextFormContainer` }
      >
        <form
          className={ `TextForm` }
        >
          <input
            className={ `WordTextInput` }
            type="text"
            name="words"
            value={ this.state.words }
            onChange={ this.handleChange }
          />

          <input
            className={ `Submit` }
            type="submit"
            value="Submit"
            onClick={ this.handleSubmit }
          />
        </form>
      </div>
    )
  }
}
