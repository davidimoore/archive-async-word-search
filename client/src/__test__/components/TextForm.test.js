import React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TextForm from 'components/TextForm';

describe('TextForm', () => {
  it('generates a snapshot', () => {
    const tree = renderer.create(<TextForm/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`changes it's input value`, () => {
    const component = shallow(<TextForm/>);
    const textInput = component.find(`input[name="words"]`);
    const inputValue = 'hat';
    textInput.simulate('change', { target: { value: inputValue } });
    const componentState = component.state();

    expect(componentState.words).toEqual(inputValue)
  })
});
