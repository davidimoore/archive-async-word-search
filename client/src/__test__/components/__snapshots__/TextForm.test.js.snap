// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextForm generates a snapshot 1`] = `
<div>
  <form>
    <input
      name="words"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
